### $\bullet$ SUM OF SQUARES (REPEATED MEASURES)
$SS_{total}=SS_{between}+SS_{within}$

$SS_{total}=SS_{between}+SS_{subject}+SS_{error}$

$SS_{within}=SS_{subject}+SS_{error}$

### $\bullet$ WORD, DEFINITION, AND COMPUTATION FORMULAS FOR SS TERMS (REPEATED-MEASURES ANOVA)
$\bullet$ $\bullet$ For the total sums of squares, $\bullet$ $\bullet$

$SS_{total}$ = the sum of squared deviations for scores about the grand mean

$=\Sigma(X-\overline{X}_{grand})^2$

$SS_{total}=\Sigma{X}^2-\dfrac{G^2}{N}$, where $G$ is the grand total and $N$ is its sample size.

$\bullet$ $\bullet$ For the between sum of squares, $\bullet$ $\bullet$

$SS_{between}$ = the sum of squared deviations for group means about the grand mean

$=n\Sigma(\overline{X}_{group}-\overline{X}_{grand})^2$

$SS_{between}=\Sigma{\dfrac{T^2}{n}}-\dfrac{G^2}{N}$, where $T$ is the group total and $n$ is its sample size (and also the number of subjects).

$\bullet$ $\bullet$ For the within sum of squares, $\bullet$ $\bullet$

$SS_{subject}=$ the sum of squared deviations of subject means about the grand mean

$=k\Sigma(\overline{X}_{subject}-\overline{X}_{grand})^2$

$SS_{subject}=\Sigma{\dfrac{T^2_{subject}}{k}}-\dfrac{G^2}{N}$, where $T_{subject}$ is the total for each subject and $k$ equals the number of repeated measures (and also the number of levels of the independent variable)

$\bullet$ $\bullet$ For the error sum of squares, $\bullet$ $\bullet$

$SS_{error}=SS_{within}-SS_{subject}$

REMINDER:

$X=score$

$T=group~total$

$n = group~sample~size;~number~of~subjects$

$G = grand~total$

$N = grand~(combined)~sample~size$

$T_{subject} = subject~total$

$k = number~of~repeated~measures$

#### $\bullet$ FORMULAS FOR df TERMS: REPEATED-MEASURES ANOVA

$df_{total}=N-1$, the number of all scores$-$1

$df_{between}=k-1$, the number of repeated measures (or levels of the independent variable)$-$1

$df_{within}=N-k$, the number of all scores$-$number of levels

$df_{subject}=n-1$, the number of subjects$-$1

$df_{error}=df_{within}-df_{subject}=(N-k)-(n-1)$

#### $\bullet$ PROPORTION OF EXPLAINED VARIANCE (REPEATED MEASURES)

$\eta^2_p=\dfrac{SS_{between}}{SS_{total}-SS_{subject}}=\dfrac{SS_{between}}{(SS_{between}+SS_{subject}+SS_{error})-SS_{subject}}$

$\eta^2_p=\dfrac{SS_{between}}{SS_{between}+SS_{error}}$

where $\eta^2_p$ is referred to as a partial $\eta^2$ or more technically as a partial squared curvilinear correlation

$\bullet$ Guidelines for $\eta^2_p$

\begin{array}{ccc}
\text{$\eta^2_p$} & \text{EFFECT} &\\
\hline
.01 & small \\
.09 & medium \\
.25 & large \\
\end{array}

#### $\bullet$ TUKEY’S HSD TEST (REPEATED MEASURES)

$HSD=q\sqrt{\dfrac{MS_{error}}{n}}$

$\bullet$ to obtain q from the table use df_error and k

#### $\bullet$ STANDARDIZED EFFECT SIZE, COHEN’S d (ADAPTED FOR REPEATED-MEASURES ANOVA)

$d=\dfrac{\overline{X}_1-\overline{X}_2}{\sqrt{s^2_p}}=\dfrac{\overline{X}_1-\overline{X}_2}{\sqrt{MS_{error}}}$

In [15]:
from IPython.display import Image
image_path = "stat_infographic/Anova_repeated_measues.png"
Image(filename=image_path)

FileNotFoundError: [Errno 2] No such file or directory: 'stat_infographic/Anova_repeated_measues.png'

In [7]:
# Sample For
import statistics
b = [[0,4,2],[3,6,6],[6,8,10]]
final = []
for index, item in enumerate(b):
    # print(index)
    temp = []
    final.append(temp)
    for n in range(len(b)):
        # print(f'n = {n}')
        # print(b[n][index])
        temp.append(b[n][index])
    print(f'temp = {temp}')
print(f'final = {final}')

mean = [statistics.mean(list) for list in final]
print(f'mean = {mean}')

# Another way using while an for loop
silence = [6,11,5,7,4,10]
white_noise = [4,3,4,6,6,7]
rock = [2,0,1,2,4,5]
the_data = [silence,white_noise,rock]
# list = [[0,4,2],[3,6,6],[6,8,10]]
final = []
n = 0
while n < len(the_data[0]):
    # print(f'n = {n}')
    temp = []
    final.append(temp)
    for i in range(len(the_data)):
        # print(f'i = {i}')
        # print(list[i][n])
        temp.append(the_data[i][n])
    # print(temp)
    n+=1
    # print('+++++')
print(final)

temp = [0, 3, 6]
temp = [4, 6, 8]
temp = [2, 6, 10]
final = [[0, 3, 6], [4, 6, 8], [2, 6, 10]]
mean = [3, 6, 6]
[[6, 4, 2], [11, 3, 0], [5, 4, 1], [7, 6, 2], [4, 6, 4], [10, 7, 5]]


In [2]:
def SS_Terms_Repeated_ANOVA(the_data):
    import statistics,math,numpy

    joined_data = numpy.concatenate(the_data)
    print(f'joined_data = {joined_data}')

    k = len(the_data)
    print(f'k = {k}')

    N = len(joined_data)
    print(f'N = {N}')

    n = len(the_data[0])
    print(f'n = {n}')

    df_total = N-1
    print(f'df_total = {df_total}')

    df_between = k-1
    print(f'df_between = {df_between}')

    df_within = N-k
    print(f'df_within = {df_within}')

    df_subject = n-1
    print(f'df_subject = {df_subject}')

    df_error = df_within - df_subject
    print(f'df_error = {df_error}')
    
    T = [sum(data) for data in the_data]
    print(f'T = {T}')

    G = sum(T)
    print(f'G = {G}')

    T_sq_over_n = [t**2/n for t in T]
    print(f'T_sq_over_n = {T_sq_over_n}')

    G_sq_over_N = G**2/N
    print(f'G_sq_over_N = {G_sq_over_N}')

    SS_between = sum(T_sq_over_n) - G_sq_over_N
    print(f'SS_between = {SS_between}')

    sum_X_sq = sum([num**2 for num in joined_data])
    print(f'sum_X_sq = {sum_X_sq}')

    SS_within = sum_X_sq - sum(T_sq_over_n)
    print(f'SS_within = {SS_within}')

    # Subjects
    subjects = []
    n = 0
    while n < len(the_data[0]):
        temp = []
        subjects.append(temp)
        for i in range(len(the_data)):
            temp.append(the_data[i][n])
        n+=1
    print(f'subjects = {subjects}')

    subject_totals = [sum(list) for list in subjects]
    print(f'subject_totals = {subject_totals}')

    T_sq_subject = [num**2/k for num in subject_totals]
    print(f'T_sq_subject = {T_sq_subject}')

    SS_subject = sum(T_sq_subject) - G_sq_over_N
    print(f'SS_subject = {SS_subject}')

    SS_error = SS_within - SS_subject
    print(f'SS_error = {SS_error}')

    SS_total = sum_X_sq - G_sq_over_N
    print(f'SS_total = {SS_total}')

    MS_between = SS_between/df_between
    print(f'MS_between = {MS_between}')

    MS_error = SS_error/df_error
    print(f'MS_error = {MS_error}')

    F_ratio = MS_between/MS_error
    print(f'F_ratio = {F_ratio}')

    eta_sq_p = SS_between/(SS_between + SS_error)
    print(f'proportion of explained variance = {eta_sq_p}')

    means = [statistics.mean(list) for list in the_data]
    print(f'means = {means}')

X0 = [0,4,2]
X24 = [3,6,6]
X48 = [6,8,10]
a = [X0,X24,X48]
SS_Terms_Repeated_ANOVA(a)

# At .05 significance level, df_error = 4, k = 3, q=5.04
n = 3
q=5.04
MS_error = 1.0
HSD = q*(MS_error/n)**0.5
print(f'HSD = {HSD}')

# Using SLEEP-DEPRIVATION EXPERIMENT: REPEATED MEASURES (means)
d48_0 = 6/(MS_error**0.5)
d48_24 = 3/(MS_error**0.5)
print(f'd48_0 = {d48_0}\nd48_24 = {d48_24}')

joined_data = [ 0  4  2  3  6  6  6  8 10]
k = 3
N = 9
n = 3
df_total = 8
df_between = 2
df_within = 6
df_subject = 2
df_error = 4
T = [6, 15, 24]
G = 45
T_sq_over_n = [12.0, 75.0, 192.0]
G_sq_over_N = 225.0
SS_between = 54.0
sum_X_sq = 301
SS_within = 22.0
subjects = [[0, 3, 6], [4, 6, 8], [2, 6, 10]]
subject_totals = [9, 18, 18]
T_sq_subject = [27.0, 108.0, 108.0]
SS_subject = 18.0
SS_error = 4.0
SS_total = 76.0
MS_between = 27.0
MS_error = 1.0
F_ratio = 27.0
proportion of explained variance = 0.9310344827586207
means = [2, 5, 8]
HSD = 2.9098453567157136
d48_0 = 6.0
d48_24 = 3.0


#### $\bullet$ REPORTS IN THE LITERATURE (From sleep deprivation experiment)

Mean aggression scores of 2, 5, and 8 were obtained when the same subjects were exposed to 0, 24, and 48 hours of sleep deprivation, respectively. There is evidence that, on average, aggression scores increase with hours of sleep deprivation [F (2, 4) = 27, MSE = 1.0, p < .01, $\eta^2_p$ = .93]. According to Tukey’s HSD test, all pairs of differences were significant (HSD = 2.87, p < .05 with 3 ≤ d ≤ 6).

ALL POSSIBLE ABSOLUTE DIFFERENCES BETWEEN PAIRS OF MEANS

SLEEP-DEPRIVATION EXPERIMENT: REPEATED MEASURES
\begin{array}{ccc}
\ & \overline{X}_0=2 & \overline{X}_{24}=5 & \overline{X}_{48}=8 \\
\hline
\overline{X}_0=2 & - & 3 & 6 \\
\overline{X}_{24}=5 &  & - & 3 \\
\overline{X}_{48}=8 &  &  & - \\
\end{array}

#### Progress Check *17.2 A school psychologist tests the effects of environmental noises on the reading comprehension scores of high school students who rotate, with the customary controls, through three different conditions: silence, white noise, and rock music. The reading comprehension scores for six subjects are as follows:

\begin{array}{ccc}
\text{SUBJECT} & \text{SILENCE} & \text{WHITE NOISE} & \text{ROCK} & \text{T}_{subject} \\
\hline
A & 6 & 4 & 2 & 12 \\
B & 11 & 3 & 0 & 14 \\
C & 5 & 4 & 1 & 10 \\
D & 7 & 6 & 2 & 15 \\
E & 4 & 6 & 4 & 14 \\
F & 10 & 7 & 5 & 22 \\
Total & 43 & 30 & 14  \\
\end{array}

#### Progress Check *17.4 
(a) Since the null hypothesis was rejected in Question 17.2, use Tukey’s HSD test to identify which pairs of population means differ significantly at the .05 level, given that the means for silence, white noise, and rock equal 7.17, 5.00, and 2.33, respectively.

(b) Use Cohen’s d to estimate the effect size for any statistically significant pairs of observed means.

(c) Interpret the results. The partial eta-squared, $\eta^2_p$, equals .64, a large effect. Mean reading comprehension is significantly higher when silence is compared with rock, with a standardized effect size. d, equivalent to almost two and one-half standard deviations. (proportion of explained variance = 0.6355421686746994)

In [3]:
silence = [6,11,5,7,4,10]
white_noise = [4,3,4,6,6,7]
rock = [2,0,1,2,4,5]
list = [silence,white_noise,rock]

# a)
SS_Terms_Repeated_ANOVA(list)
# At df_between = 2, df_error = 10, .05 level of significance, the critical value of F = 4.10, which is less than the obsereved value of
# F_ratio = 8.71900826446283, therefore we reject the null hypothesis

# b)
MS_error = 4.033333333333326
d_silence_rock = 4.833333333/(MS_error**0.5)
d_whitenoise_rock = 2.666666667/(MS_error**0.5)
print(f'd_silence_rock = {d_silence_rock}\nd_whitenoise_rock = {d_whitenoise_rock}')

joined_data = [ 6 11  5  7  4 10  4  3  4  6  6  7  2  0  1  2  4  5]
k = 3
N = 18
n = 6
df_total = 17
df_between = 2
df_within = 15
df_subject = 5
df_error = 10
T = [43, 30, 14]
G = 87
T_sq_over_n = [308.1666666666667, 150.0, 32.666666666666664]
G_sq_over_N = 420.5
SS_between = 70.33333333333337
sum_X_sq = 559
SS_within = 68.16666666666663
subjects = [[6, 4, 2], [11, 3, 0], [5, 4, 1], [7, 6, 2], [4, 6, 4], [10, 7, 5]]
subject_totals = [12, 14, 10, 15, 14, 22]
T_sq_subject = [48.0, 65.33333333333333, 33.333333333333336, 75.0, 65.33333333333333, 161.33333333333334]
SS_subject = 27.83333333333337
SS_error = 40.33333333333326
SS_total = 138.5
MS_between = 35.166666666666686
MS_error = 4.033333333333326
F_ratio = 8.71900826446283
proportion of explained variance = 0.6355421686746994
means = [7.166666666666667, 5, 2.3333333333333335]
d_silence_rock = 2.4066597222052097
d_whitenoise_rock = 1.3278122607845622


\begin{array}{ccc}
\ & \overline{X}_{silence}=7.166666666666667 & \overline{X}_{white~noise}=5 & \overline{X}_{rock}=2.3333333333333335 \\
\hline
\overline{X}_{silence}=7.166666666666667 & - & - & - \\
\overline{X}_{white~noise}=5 & 2.166666667 & - & - \\
\overline{X}_{rock}=2.3333333333333335 & 4.833333333 & 2.666666667 & - \\
\end{array}

#### *17.5 Capitalizing on the additivity of SS and df terms, complete the following ANOVA summary table, assuming repeated measures for 12 subjects across four levels of the independent variable and the .05 level of significance.
$SS_{within}$ = 8800; $SS_{subject}$ = 5500; Total = 10000

find:

$all~dfs,~SS_{between},~SS_{error},~MS_{between},~MS_{error},~F$

$SS_{error}=SS_{within}-SS_{subject}$

#### $\bullet$ FORMULAS FOR df TERMS: REPEATED-MEASURES ANOVA

$df_{total}=N-1$, the number of all scores$-$1

$df_{between}=k-1$, the number of repeated measures (or levels of the independent variable)$-$1

$df_{within}=N-k$, the number of all scores$-$number of levels

$df_{subject}=n-1$, the number of subjects$-$1

$df_{error}=df_{within}-df_{subject}=(N-k)-(n-1)$


$SS_{total}=SS_{between}+SS_{subject}+SS_{error}$

$SS_{within}=SS_{subject}+SS_{error}$

In [4]:
n = 12
k = 4
N = 48
SS_total = 10000
SS_within = 8800
SS_subject = 5500
SS_error = SS_within - SS_subject
df_total = N - 1
df_between = k - 1
df_within = N - k
df_subject = n - 1
df_error = df_within - df_subject
SS_between = SS_total - SS_within
MS_between = SS_between/df_between
SS_error = SS_within - SS_subject
MS_error = SS_error/df_error
F_ratio = MS_between/MS_error
print(f'df_total = {df_total}')
print(f'df_between = {df_between}')
print(f'df_within = {df_within}')
print(f'df_subject = {df_subject}')
print(f'df_error = {df_error}')
print(f'SS_between = {SS_between}')
print(f'SS_error = {SS_error}')
print(f'MS_between = {MS_between}')
print(f'MS_error = {MS_error}')
print(f'F_ratio = {F_ratio}')

df_total = 47
df_between = 3
df_within = 44
df_subject = 11
df_error = 33
SS_between = 1200
SS_error = 3300
MS_between = 400.0
MS_error = 100.0
F_ratio = 4.0


#### *17.6 Return to the study first described in Question 16.5 on page 308, where a psychologist tests whether shy college students initiate more eye contacts with strangers because of training sessions in assertive behavior. Use the same data, but now assume that eight subjects, coded as A, B, . . . G, H, are tested repeatedly after zero, one, two, and three training sessions. (Incidentally, since the psychologist is interested in any learning or sequential effect, it would not make sense—indeed, it’s impossible, given the sequential nature of the independent variable—to counterbalance the four sessions.) The results are expressed as the observed number of eye contacts:

WORKSHOP SESSIONS
\begin{array}{ccc}
\text{SUBJECT} & \text{ZERO} & \text{ONE} & \text{TWO} & T_{subject} \\
\hline
A &1 & 2 & 4 & 14 \\
B &0 & 1 & 2 & 9 \\
C &0 & 2 & 3 & 11 \\
D &2 & 4 & 6 & 19 \\
E &3 & 4 & 7 & 23 \\
F &4 & 6 & 8 & 28 \\
G &2 & 3 & 5 & 18 \\
H &1 & 3 & 5 & 16 \\
\end{array}

(a) Summarize the results with an ANOVA table. Short-circuit computational work by using the results in Question 16.5 for the SS terms, that is, $SS_{between} = 154.12, SS_{within} = 132.75,~and~SS_{total} = 286.87$.

(b) Whenever appropriate, estimate effect sizes with $\eta_p^2$ and with d, and conduct Tukey’s HSD test.

(c) Compare these results with repeated measures with those in Question 16.5 for independent samples.

#### $\bullet$ TUKEY’S HSD TEST (REPEATED MEASURES)

$HSD=q\sqrt{\dfrac{MS_{error}}{n}}$

$\bullet$ to obtain q from the table use df_error and k

#### $\bullet$ STANDARDIZED EFFECT SIZE, COHEN’S d (ADAPTED FOR REPEATED-MEASURES ANOVA)

$d=\dfrac{\overline{X}_1-\overline{X}_2}{\sqrt{s^2_p}}=\dfrac{\overline{X}_1-\overline{X}_2}{\sqrt{MS_{error}}}$

In [5]:
zero = [1,0,0,2,3,4,2,1]
one = [2,1,2,4,4,6,3,3]
two = [4,2,3,6,7,8,5,5]
three = [7,1,6,9,10,12,8,7]
list = [zero,one,two,three]
SS_Terms_Repeated_ANOVA(list)

# At .05 significance level df_error = 21, df_between = 3, the critical value of F = 3.07, the observed value F_ratio = 52.95092024539878,
# reject the null hypothesis

joined_data = [ 1  0  0  2  3  4  2  1  2  1  2  4  4  6  3  3  4  2  3  6  7  8  5  5
  7  1  6  9 10 12  8  7]
k = 4
N = 32
n = 8
df_total = 31
df_between = 3
df_within = 28
df_subject = 7
df_error = 21
T = [13, 25, 40, 60]
G = 138
T_sq_over_n = [21.125, 78.125, 200.0, 450.0]
G_sq_over_N = 595.125
SS_between = 154.125
sum_X_sq = 882
SS_within = 132.75
subjects = [[1, 2, 4, 7], [0, 1, 2, 1], [0, 2, 3, 6], [2, 4, 6, 9], [3, 4, 7, 10], [4, 6, 8, 12], [2, 3, 5, 8], [1, 3, 5, 7]]
subject_totals = [14, 4, 11, 21, 24, 30, 18, 16]
T_sq_subject = [49.0, 4.0, 30.25, 110.25, 144.0, 225.0, 81.0, 64.0]
SS_subject = 112.375
SS_error = 20.375
SS_total = 286.875
MS_between = 51.375
MS_error = 0.9702380952380952
F_ratio = 52.95092024539878
proportion of explained variance = 0.8832378223495702
means = [1.625, 3.125, 5, 7.5]


ALL POSSIBLE ABSOLUTE DIFFERENCES BETWEEN PAIRS OF MEANS
\begin{array}{ccc}
\ & \overline{X}_0=1.625 & \overline{X}_{1}=3.125 & \overline{X}_{2}=5.00 & \overline{X}_{3}=7.50 \\
\hline
\overline{X}_0=1.625 & - & 1.5 & 3.375 & 5.875 \\
\overline{X}_{1}=3.125 &  & - & 1.875 & 4.375 \\
\overline{X}_{2}=5.00 &  &  & - & 2.5 \\
\overline{X}_{3}=7.50 &  &  &  & - \\
\end{array}

In [6]:
# c
MS_error = 0.9702380952380952
dX3X0 = 5.875/MS_error**0.5
dX3X1 = 4.375/MS_error**0.5
dX2X0 = 3.375/MS_error**0.5
dX3X2 = 5.875/MS_error**0.5
print(f'dX3X0 = {dX3X0}\ndX3X1 = {dX3X1}\ndX2X0 = {dX2X0}\ndX3X2 = {dX3X2}')

dX3X0 = 5.964426753935997
dX3X1 = 4.441594391228934
dX2X0 = 3.4263728160908915
dX3X2 = 5.964426753935997


$\approx$